### PR TITLE
Added c++ solution for Mooyoo Mooyoo (USACO 860)

### PR DESCRIPTION
**Pull request by pranavgithub1**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
> I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following:

are-you-sure-john-cena

**Comment by SansPapyrus683** [comment]
image

**Comment by pranavgithub1** [comment]
I am stupid apparently and forgot to select anything when I used the 'covert indentation to tabs' command in the editor lol.

**Comment by pranavgithub1** [comment]
I retested the code on USACO.

File: solutions/usaco-860.mdx [hunk 1]
```diff
@@ -13,6 +13,128 @@ author: Nathan Gong
...
+
+// stores current component indices to set to '0'
+vector<pair<int, int>> component;
+
+
+/*
+	floodfill to count connected regions
+	returns size of connected region
+	fills 'component' vector with the indices of the component
+*/
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:36]:
```cpp
/*
 * format multiline
 * comments something like this
 */
```

File: solutions/usaco-860.mdx [hunk 2]
```diff
@@ -13,6 +13,128 @@ author: Nathan Gong
...
+		board[i][j] != color
+	) {
+		return 0;
+	}
+	visited[i][j] = 1;
+	component.push_back({i, j});
+	cnt += floodfill(i + 1, j, color)
+	+ floodfill(i - 1, j, color)
+	+ floodfill(i, j + 1, color)
+	+ floodfill(i, j - 1, color);
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:51]:
```suggestion
	cnt += (
		floodfill(i + 1, j, color)
		+ floodfill(i - 1, j, color)
		+ floodfill(i, j + 1, color)
		+ floodfill(i, j - 1, color)
	);
```

File: solutions/usaco-860.mdx [hunk 3]
```diff
@@ -13,6 +13,131 @@ author: Nathan Gong
...
 
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int n, k;
+const int maxRows = 100;
+const int maxCols = 10;
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:24]: rename to `MAX_ROWS` and `MAX_COLS`

File: solutions/usaco-860.mdx [hunk 4]
```diff
@@ -13,6 +13,131 @@ author: Nathan Gong
...
+
+/*
+ * floodfill to count connected regions
+ * returns size of connected region
+ * fills 'component' vector with the indices of the component
+ */
+int floodfill(int i, int j, char color) {
+	int cnt = 1;
+	if (
+		i < 0 || j < 0 || i >= n || j >= 10 ||
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:40]: what does 10 mean?

File: solutions/usaco-860.mdx [hunk 5]
```diff
@@ -13,6 +13,131 @@ author: Nathan Gong
...
+int main() {
+	freopen("mooyomooyo.in", "r", stdin);
+	freopen("mooyomooyo.out", "w", stdout);
+
+	cin >> n >> k;
+
+	for (int i = n - 1; i >= 0; i--) {
+		string nums;
+		cin >> nums;
+		for (int j = 0; j < 10; j++) {
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:67]: what's this 10?
other places too

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Comment by pranavgithub1** [comment]
The 10s are MAX_COLS

File: solutions/usaco-860.mdx [hunk 6]
```diff
@@ -13,6 +13,131 @@ author: Nathan Gong
...
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int n, k;
+const int MAX_ROWS = 100;
+const int MAX_COLS = 10;
+char board[MAX_ROWS][MAX_COLS];
+vector<vector<bool>> visited(MAX_ROWS, vector<bool>(MAX_COLS, false));
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:26]: wow can't believe you actually took my suggestion
but uh
```suggestion
vector<vector<bool>> visited(MAX_ROWS, vector<bool>(MAX_COLS));
```
the default value is false jsyk

File: solutions/usaco-860.mdx [hunk 7]
```diff
@@ -13,6 +13,131 @@ author: Nathan Gong
...
+			board[i][j] = nums[j];
+		}
+	}
+
+	/*
+	 * Loop while there are regions that got destroyed (board changed)
+	 * If no regions were destroyed, we are at the final state (break loop; return answer)
+	 */
+	bool boardChanged = true;
+	while (boardChanged) {
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:77]: (optional) most of the c++ here uses snake_case, so maybe use that in your own code as well?

**Review by SansPapyrus683 — APPROVED** [review]
optional stuff

**Review by maggie-j-liu — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
if the build works i'll merge

File: solutions/usaco-860.mdx [hunk 8]
```diff
@@ -125,7 +250,7 @@ public class MooyoMooyo {
 		return size;
 	}
 
-	CodeSnip{Kattio}
+	//CodeSnip{Kattio}
```
Comment (nathangong) [inline solutions/usaco-860.mdx:253]:
```suggestion
	// CodeSnip{Kattio}
```

File: solutions/usaco-860.mdx [hunk 9]
```diff
@@ -13,6 +13,131 @@ author: Nathan Gong
...
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int n, k;
+const int MAX_ROWS = 100;
+const int MAX_COLS = 10;
+char board[MAX_ROWS][MAX_COLS];
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:25]: oh uh if you changed `visited` to a vector could you do the same for `board` as well?

**Comment by SansPapyrus683** [comment]
uh sorry wait i'll merge after this build has finished

**Comment by pranavgithub1** [comment]
Is there anything else I need to do?

**Comment by SansPapyrus683** [comment]
no, we chillin :heart:

**Pull request merged**